rocAL/rocAL_pybind/amd/rocal/plugin/pytorch.py

```diff
@@ -35,90 +35,98 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
         self.device = device
         self.device_id = device_id
         self.batch_size = self.loader._batch_size
-        self.output = self.dimensions = self.torch_dtype = None
+        self.labels_size = ((self.batch_size * self.loader._num_classes) if self.loader._one_hot_encoding else self.batch_size)
+        self.output_list = self.dimensions = self.torch_dtype = None
+        self.output_memory_type = self.loader._output_memory_type
         self.len = b.getRemainingImages(self.loader._handle)
         self.display = display
+        if self.loader._name is None:
+            self.loader._name = self.loader._reader
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        if(b.isEmpty(self.loader._handle)):
+        if self.loader.rocalRun() != 0:
```

Comment:
rocalRun doesn't check for empty images and it returns non-zero values in case of empty loader. Is the empty check still needed if we can use rocalRun for the same check?